hilti/include/rt/types/map.h

```diff
@@ -141,6 +141,51 @@ class ConstIterator {
         : _control(control), _iterator(std::move(iterator)) {}
 };
 
+namespace detail {
+
+/** Proxy class for performing save assignments to `Map` entries.
+ *
+ * @note All method accessing the underlying map are only defined for r-values
+ * since this class only holds a reference to it. That makes it in general
+ * unsafe to use these methods when the instance was bound to a later expired
```

Comment:
That leads to the question: are they situations where generated code could end up using this in an unsafe way? 

Should we disable copy/move ctor and self-assignment operators to enforce that one can't do much with an AssignProxy?

Alternative would be using control blocks here too.

Reply:
I think the code we generate should be safe. I think there is no way in C++ to disallow users to bind a name to instances of this class. The assignment operators should not be generated here as the class contains a reference member which cannot be reseated.

Re:control block, that was my first attempt, but I ran into issues in constructs like `map["entry"] = "foo"` which do not involve any sequence points between construction and destruction of the proxy which lead to control blocks not being destructed as I naively expected.